spark-operator/src/main/java/org/apache/spark/kubernetes/operator/config/SparkOperatorConf.java

```diff
@@ -0,0 +1,407 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.spark.kubernetes.operator.config;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import io.javaoperatorsdk.operator.api.config.LeaderElectionConfiguration;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+
+import org.apache.spark.kubernetes.operator.listeners.ApplicationStatusListener;
+
+import static org.apache.spark.kubernetes.operator.reconciler.SparkReconcilerUtils.defaultOperatorConfigLabels;
+import static org.apache.spark.kubernetes.operator.reconciler.SparkReconcilerUtils.labelsAsStr;
+
+/**
+ * Spark Operator Configuration options.
+ */
+@Slf4j
+public class SparkOperatorConf {
+  public static final String PREFIX = "spark.operator.";
+  public static final String METRIC_PREFIX = "spark.metrics.conf.operator.";
+  public static final String SINK = "sink.";
+  public static final String CLASS = "class";
+
+  public static final ConfigOption<String> OperatorAppName = ConfigOption.<String>builder()
+      .key(PREFIX + "name")
```

Comment:
suggestion: always use the plain string for configuration definition, because users/developers are likely to search a configuration key in IDE globally

Reply:
https://github.com/apache/spark-kubernetes-operator/pull/2/commits/bae3e140f20b6e227b3eab6691bbb7675314bc88 fixes this